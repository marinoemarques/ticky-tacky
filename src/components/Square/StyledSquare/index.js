import styled from 'styled-components'

import {
  PLAYER_ONE,
  TACKY_COLOUR,
  TICKY_COLOUR
} from '../../../constants'

function getColour ({ isLoser, player }) {
  if (isLoser) { return 'Gainsboro' }

  return player === PLAYER_ONE
    ? TICKY_COLOUR
    : TACKY_COLOUR
}

const StyledSquare = styled.span`
  align-self: stretch;
  background-color: White;
  color: ${getColour};;
  cursor: ${props => props.onClick ? 'pointer' : 'not-allowed'};
  font-size: 16vh;
  font-weight: bold;
  grid-area: ${props => props.area || '0'};
  justify-self: stretch;
  text-align: center;
  text-transform: uppercase;
`
StyledSquare.displayName = 'StyledSquare'

export default StyledSquare
