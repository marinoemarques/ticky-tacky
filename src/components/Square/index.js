import * as React from 'react'

import styled from 'styled-components'
import styledMap from 'styled-map'

const { useState } = React

const StyledSquare = styled.span`

  border: 1px solid LightGray;
  color: ${styledMap('player', {
    x: 'Crimson',
    default: 'SlateBlue'
  })};;
  cursor: ${props => props.onClick ? 'pointer' : 'not-allowed'};
  display: inline-block;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  min-height: 21vh;
  min-width: 21vh;
  text-align: center;
  text-transform: uppercase;
`
StyledSquare.displayName = 'StyledSquare'

export default function Square () {
  const [player, setPlayer] = useState('')

  return (
    <StyledSquare onClick={() => setPlayer(player === 'x' ? 'o' : 'x')}
      player={player}>
      {player}</StyledSquare>
  )
}
