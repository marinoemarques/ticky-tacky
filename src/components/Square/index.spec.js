import * as React from 'react'

import Square from './'
import { create } from 'react-test-renderer'

describe('Square', () => {
  it('matches the snapshots', () => {
    const testInstance = create(<Square />)
    const square = testInstance.root.findByType('span')

    expect(testInstance.toJSON()).toMatchSnapshot()
    square.props.onClick()
    expect(testInstance.toJSON()).toMatchSnapshot()
    square.props.onClick()
    expect(testInstance.toJSON()).toMatchSnapshot()
  })
})
