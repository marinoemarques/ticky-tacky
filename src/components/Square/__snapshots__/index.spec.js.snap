// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square matches the snapshots 1`] = `
.c0 {
  border: 1px solid LightGray;
  color: SlateBlue;
  cursor: pointer;
  display: inline-block;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  min-height: 21vh;
  min-width: 21vh;
  text-align: center;
  text-transform: uppercase;
}

<span
  className="c0"
  onClick={[Function]}
>
  
</span>
`;

exports[`Square matches the snapshots 2`] = `
.c0 {
  border: 1px solid LightGray;
  color: Crimson;
  cursor: pointer;
  display: inline-block;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  min-height: 21vh;
  min-width: 21vh;
  text-align: center;
  text-transform: uppercase;
}

<span
  className="c0"
  onClick={[Function]}
>
  x
</span>
`;

exports[`Square matches the snapshots 3`] = `
.c0 {
  border: 1px solid LightGray;
  color: SlateBlue;
  cursor: pointer;
  display: inline-block;
  font-size: 16vh;
  font-weight: bold;
  line-height: 20vh;
  min-height: 21vh;
  min-width: 21vh;
  text-align: center;
  text-transform: uppercase;
}

<span
  className="c0"
  onClick={[Function]}
>
  o
</span>
`;
